lib/devise-security/models/password_expirable.rb

```diff
@@ -4,66 +4,83 @@
 
 module Devise
   module Models
-
-    # PasswordExpirable takes care of change password after
+    # PasswordExpirable makes passwords expire after a configurable amount of
+    # time, or on demand
     module PasswordExpirable
       extend ActiveSupport::Concern
 
       included do
         before_save :update_password_changed
       end
 
-      # is an password change required?
+      # Is a password change required?
+      # @return [Boolean]
+      # @return [true] if +password_changed_at+ has not been set or if it is old
+      #   enough based on +expire_password_after+ configuration.
       def need_change_password?
-        if expired_password_after_numeric?
-          self.password_changed_at.nil? || self.password_changed_at < self.expire_password_after.seconds.ago
+        if enabled?
+          if expire_on_demand?
+            # Only expire when password_changed_at is nil
+            password_changed_at.nil?
+          else
+            # Expire when password_changed_at is nil or when the last time the
+            # password was changed is sufficiently old
+            password_changed_at.nil? || password_changed_at < expire_password_after.seconds.ago
```

Comment:
Possibly extract this line into a private method to address the CC issue? It's borderline to me.